packages/owl-bot/src/bin/commands/enqueue-copy-tasks.ts

```diff
@@ -0,0 +1,149 @@
+
+interface Args {
+  'pem-path': string;
+  'app-id': number;
+  installation: number;
+  repo: string;
+  'git-path': string;
+  project: string;
+  'firestore-project': string;
+  queue: string;
+  trigger?: string;
+}
+
+export const enqueueCopyTasks: yargs.CommandModule<{}, Args> = {
+  command: 'enqueue-copy-tasks',
+  describe:
+    'look at most recent commits to repo and enqueue copy tasks for those repos',
+  builder(yargs) {
+    return yargs
+      .option('pem-path', {
+        describe: 'provide path to private key for requesting JWT',
+        type: 'string',
+        demand: true,
+      })
+      .option('app-id', {
+        describe: 'GitHub AppID',
+        type: 'number',
+        demand: true,
+      })
+      .option('installation', {
+        describe: 'installation ID for GitHub app',
+        type: 'number',
+        demand: true,
+      })
+      .option('repo', {
+        describe: 'repository to monitor for changes',
+        type: 'string',
+        default: 'googleapis/googleapis-gen',
+      })
+      .option('git-path', {
+        describe: 'path on disk of repo to monitor for changes',
+        type: 'string',
+        demand: true,
+      })
+      .option('project', {
+        describe: 'gcloud project',
+        type: 'string',
+        default: 'repo-automation-bots',
+      })
+      .option('firestore-project', {
+        describe: 'project used for firestore database',
+        type: 'string',
+        default: 'repo-automation-bots-metrics',
+      })
+      .option('queue', {
+        describe: 'queue to publish PR update jobs to',
+        type: 'string',
+        default: 'projects/repo-automation-bots/topics/owlbot-prs',
+      })
+      .option('trigger', {
+        describe:
+          'enqueue copy job trigger (if this should be run in background)',
+        type: 'string',
+      });
+  },
+  async handler(argv) {
+    const privateKey = await readFileAsync(argv['pem-path'], 'utf8');
+    if (!privateKey) throw Error('pem-path or private-key must be provided');
+    if (argv.trigger) {
+      await triggerEnqueueCopyJobs({
+        privateKey,
+        appId: argv['app-id'],
+        installation: argv.installation,
+        project: argv.project,
+        firestoreProject: argv['firestore-project'],
+        queue: argv.queue,
+        trigger: argv.trigger,
+      });
+    } else {
+      const token = await getGitHubShortLivedAccessToken(
+        privateKey,
+        argv['app-id'],
+        argv.installation
+      );
+      const octokit = await getAuthenticatedOctokit(token.token);
+      admin.initializeApp({
+        credential: admin.credential.applicationDefault(),
+        projectId: argv['firestore-project'],
+      });
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const db = admin.firestore();
+      const configStore = new FirestoreConfigsStore(db!);
+
+      const shas = await getLastNCommits(argv.repo, octokit, 1);
```

Comment:
I was figuring we'd start by looking at the last commit until we've bootstrapped the repository, i.e., have a history of commits, at which point we can pull some reasonable number of previous commits to look back in time, and ensure we're up-to-date.

Reply:
What's the goal?

Reply:
I was thinking about the task we schedule that walks commits every hour or so, looking for any we missed. It seemed like if we pick a reasonable number of commits to look back in time, e.g., 25:

* we can backfill 25 commits to make sure that they're in the DB.
* always check 25 commits when we're ensuring we're caught up.

Reply:
I've refactored this logic to use an infinite async iterator instead.